src/auth/idtokenclient.ts

```diff
@@ -41,7 +46,7 @@ export class IdTokenClient extends OAuth2Client {
    * See: https://developers.google.com/compute/docs/authentication
    */
   constructor(options: IdTokenOptions) {
-    super();
+    super(options);
```

Comment:
This is new functionality right? Looks like the IdTokenClient didn't previously support RefreshClient options. Are we sure it should be used here?

Reply:
I thought it would be cleaner to pass the parameters directly to it's parent; however we mostly care about setting `AuthClientOptions`. I figured it wouldn't hurt to expose underlying `OAuth2ClientOptions`, as it wouldn't harm `IdTokenClient`'s functionality. Thoughts?

Also, I noticed the `OAuth2ClientOptions` interface was accidentally duplicated from a merge; removed the duplicated interface to avoid confusion.